src/prefect/client/orchestration/_concurrency_limits/client.py

```diff
@@ -649,6 +778,27 @@ async def release_concurrency_slots(
             },
         )
 
+    async def release_concurrency_slots_with_lease(
+        self,
+        lease_id: "UUID",
+        occupancy_seconds: float,
+    ) -> "Response":
+        """
+        Release concurrency slots for the specified lease.
+
+        Args:
+            lease_id: The ID of the lease corresponding to the concurrency limits to release.
+            occupancy_seconds: The duration in seconds that the slots were occupied.
```

Comment:
this looks weird to see exposed as configurable - what's the idea here?

Reply:
This mirrors the existing `release_concurrency_slots` method, allowing the `concurrency` context manager to report how long it held the slots. Both the `/v2/concurrency_limits/decrement` and `/v2/concurrency_limits/decrement-with-lease` endpoints accept `occupancy_seconds`.

Reply:
`occupancy_seconds` is used by the server to inform how long to tell clients to wait when they've been denied slot acquisition.